dialogue-core/src/main/java/com/palantir/dialogue/core/QueuedChannel.java

```diff
@@ -114,8 +117,12 @@ private void onCompletion() {
      * Try to schedule as many tasks as possible. Called when requests are submitted and when they complete.
      */
     private void schedule() {
+        int numScheduled = 0;
         while (scheduleNextTask()) {
-            // Do nothing
+            numScheduled++;
+        }
+        if (numScheduled > 1) {
```

Comment:
```suggestion
        if (log.isDebugEnabled()) {
```
zero and one are also helpful data points imo.